ocean_lib/data_provider/data_service_provider.py

```diff
@@ -103,7 +102,137 @@ def encrypt(
 
     @staticmethod
     @enforce_types
-    def sign_message(wallet: Wallet, msg: str) -> str:
+    def fileinfo(did: str, service_id: str, service_endpoint: str) -> Response:
+        payload = {"did": did, "serviceId": service_id}
+
+        response = DataServiceProvider._http_method(
+            "post", service_endpoint, json=payload
+        )
+
+        if not response or not hasattr(response, "status_code"):
+            raise Exception("Response not found!")
+
+        if response.status_code != 200:
+            msg = (
+                f"Fileinfo service failed at the FileInfoEndpoint "
+                f"{service_endpoint}, reason {response.text}, status {response.status_code}"
+            )
+            logger.error(msg)
+            raise Exception(msg)
+
+        logger.info(
+            f"Retrieved asset files successfully"
+            f" FileInfoEndpoint {service_endpoint}"
+        )
+        return response
+
+    @staticmethod
+    @enforce_types
+    def initialize(
```

Comment:
by the way, initialize is a duplicate of the order_requirements function, I think. Please delete the previous version and make sure to adapt any of its usages.